src/wallet/spend.h

```diff
@@ -51,8 +51,10 @@ struct CoinsResult {
     void Shuffle(FastRandomContext& rng_fast);
     void Add(OutputType type, const COutput& out);
 
-    /** Sum of all available coins */
+    /** Sum of all available coins raw value */
     CAmount total_amount{0};
+    /** Sum of all available coins effective value (each output value minus fees required to spend it) */
+    std::optional<CAmount> total_effective_amount{0};
```

Comment:
note to self: not a huge fan of how this looks like with `optional`. The code would look much simpler if we can just always pass a fee rate to `AvailableCoins` (we can set it 0 by default). Can't make my mind which way is better...

Reply:
I would agree if we were using `AvailableCoins` only for the tx creation process, but we are using the function for other processes that don't require to calculate the effective value nor the output spending fee.

We definitely can improve it but I don't think that this PR is the best place to do it.

Side from that, from my side, have some plans to re-write the `AvailableCoins` functionality differently post #25806 (because, after it, will be able to continually connect outputs to existent output groups, outside of the coin selection process which is currently forcing us to do the whole calculation on-demand).